integration/admin-client/src/main/java/org/keycloak/admin/client/Keycloak.java

```diff
@@ -41,10 +42,27 @@
  * @see KeycloakBuilder
  */
 public class Keycloak implements AutoCloseable {
+
+    private static final ResteasyClientProvider CLIENT_PROVIDER;
+
+    static {
+        Iterator<ResteasyClientProvider> iterator = ServiceLoader.load(ResteasyClientProvider.class).iterator();
```

Comment:
How is it intended to work as you have a declaration in this module? 
If we have a second provider on the classpath, it will fail (there is a check).
If we exclude RESTEasy Classic from the classpath, it will fail to instantiate it.

One solution would be to move the declaration to its own module.

Reply:
The idea here is to avoid an additional module and force a specific implementation when running the extension build steps.

Isn't that be possible in Quarkus?

Reply:
it's not possible at all (not even related to Quarkus). 

It means that you need to be able to load the provider and its dependencies and you have a default declared one. If we want to use another provider and not ship the default one, we cannot.

Reply:
In Quarkus, we can exclude a classpath resource (in this case the service file), so this should work

Reply:
Well, yes, but that's clearly a hack and we would also need to strip all the dependencies of that implementation.
It would be so much better to fix it correctly here but move the provider into its own module.

If you do not want to break, just declare an optional dependency on it, and we can exclude that dependency in Quarkus. It would be much better than removing a service file.

Reply:
@cescoffier I see your point and makes more sense to create the default via reflection instead.